File: python/src/mergeAlternately_test.py
from typing import List

import pytest

from mergeAlternately import Solution

solution = Solution()


@pytest.mark.parametrize(
    "fn",
    (
        (solution.mergeAlternately),
        (solution.mergeAlternately_python),
    ),
)
@pytest.mark.parametrize(
    "word1, word2, want",
    (
        ("abc", "pqr", "apbqcr"),
        ("ab", "pqrs", "apbqrs"),
        ("abcd", "pq", "apbqcd"),
    ),
)
def test_mergeAlternately(fn: callable, word1: str, word2: str, want: int):
    got = fn(word1, word2)
    assert got == want

File: c/mergeAlternately_test.c
#include <assert.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

#include "mergeAlternately.c"

typedef struct test {
    char* word1;
    char* word2;
    char* want;
} test;

int main(int argc, char* argv[])
{
    test tests[] = {
        { "abc", "pqr", "apbqcr" },
        { "ab", "pqrs", "apbqrs" },
        { "abcd", "pq", "apbqcd" },
    };
    size_t len_tests = sizeof(tests) / sizeof(test);

    for (size_t i = 0; i < len_tests; i++) {
        test* t = &tests[i];
        char* got = mergeAlternately(t->word1, t->word2);
        assert(strcmp(got, t->want) == 0);
        free(got);
    }

    exit(EXIT_SUCCESS);
}
